test/k8sT/Policies.go

```diff
@@ -971,7 +985,12 @@ EOF`, k, v)
 				secondNS, netpolNsSelector, helpers.KubectlApply, helpers.HelperTimeout)
 			Expect(err).Should(BeNil(), "Policy cannot be applied")
 
+			// See GH-8781 for more context on why this is needed
+			waitServiceBackendsPlumbed()
```

Comment:
I understand why the `waitServiceBackendsPlumbed` is needed but I don't understand why this is needed on this line as well.

Reply:
Per analysis on #8781, the readiness probes to the pods were failing. It is unclear to me as to why the readiness probes started failing. One hypothesis I have is that the policy being imported causes the readiness probes to fail (I am not sure). As a result, the pods are not ready, so kubelet will remove them from the list of Endpoints backing the service. The call above (`CiliumPolicyAction`) waits for the policy to be applied for all endpoints, but it does not check the state of the pods themselves. So, we should check whether the policy being imported affects the readiness of the pods themselves. I could also wait for the pods to be ready, but I figure checking whether the Endpoints are plumbed is better per my comment above.

Reply:
> It is unclear to me as to why the readiness probes started failing. One hypothesis I have is that the policy being imported causes the readiness probes to fail (I am not sure).

This is really dangerous. We should be sure what is causing issues, we can't blindly add stuff in the CI just to make it green. If the readiness probe is failing because of policy being added we should make sure why is that happening. Does this also means when we add policy there's connectivity disruption? Maybe it is a regression and adding this line will hide that regression making it to show up later in production environments that will be harder to track it down.

Reply:
Your point is valid, I'll remove this check. 

My idea was to see if the backends were plumbed correctly before doing the connectivity check, but waiting implies that they might not be there for some time, which in this case is unacceptable. I can do that check once before we make the request to the service, as opposed to waiting for the service to be ready. This would more clearly state that the services have backends. I'll add that as a commit.

With regards to future work, I'm wondering whether we can add a check that signifies whether the readiness probe has ever failed so that we don't miss these potential momentary connectivity lapses.

Reply:
I've moved the check to the `BeforeAll` and dropped it from the test itself.